### DISCUSSION TITLE
Topo_ra.grd dimensions of range doesn't match

### DISCUSSION THREAD

**Discussion by x3zou** [gmtsar/gmtsar#676] [Q&A] [answered]
Hello,
  I'm trying to make interferograms for ALOS2 track 77 frame 2900 between 20220121 and 20230217. The area of the frame is attached below.
  The code went well for the subswath F1 and F4, and the phasefilt.grd was produced succesfully. But for F2 F3 and F5, it failed. The log file indicates that "Read topo_ra.grd .... the dimensions of range do not match" (please see attached).
  To make sure that my DEM file covers the whole area of the frame, I made a very large DEM file, which has lon=[33 41] and lat=[32 38] and I created my topo_ra.grd with the large DEM file. But the problem still persists.
  I found that the problem is that for F2 F4 and F5 subswaths, my .PRM files don't have the same dimension for the two dates. For F1 and F4, they are the same.
  Is there any command to cut my PRM files so that they have the same dimension? I tried grdcut but it doesn't support the format.
<img width="745" alt="Screenshot 2023-03-02 at 16 39 10" src="https://user-images.githubusercontent.com/103137212/222613961-c0639452-f6d5-440d-8097-417b2bb6d6ac.png">
<img width="745" alt="Screenshot 2023-03-02 at 16 38 57" src="https://user-images.githubusercontent.com/103137212/222613990-0f5eedf3-9537-4764-bdf6-0e72c8a6b567.png">

Thanks,
Xiaoyu

<img width="685" alt="Screenshot 2023-03-02 at 14 33 26" src="https://user-images.githubusercontent.com/103137212/222575518-48feda84-4586-499d-a051-bce8efa31428.png">

<img width="583" alt="Screenshot 2023-03-02 at 14 28 40" src="https://user-images.githubusercontent.com/103137212/222575334-1636bcba-0beb-45c2-8d8e-11de3eb228b0.png">

**Comment by Xiaohua-Eric-Xu** [comment, marked as answer]
Hi @x3zou , theoretically, the master and aligned images should have the same range dimension after resampling. Could you check 
the parameters in PRM files at the SLC directory, i.e. `grep num_rng_bins SLC/*PRM`. The numbers in the raw directory should not match by default. Also, please let me know if you are using `p2p_ALOS2_SCAN_Frame.csh`.

**Reply by x3zou** [reply]
Hi @Xiaohua-Eric-Xu,
  Thanks for reaching out.
  Here is the output of grep num_rng_bins SLC/*PRM for my F2 subswath.
  I was not using p2p_ALOS2_SCAN_Frame.csh. 
<img width="1032" alt="Screenshot 2023-03-06 at 11 37 30 AM" src="https://user-images.githubusercontent.com/103137212/223212932-46bc68f3-ea6b-4eec-ae39-25479ff5d8bb.png">

**Reply by Xiaohua-Eric-Xu** [reply]
Hmmm, interesting. How come that after resamp, the PRMs still have different num_rng_bins? Could you describe a bit more in details what you did to process the data within F2? 

None the less, using p2p_ALOS2_SCAN_Frame.csh may make things easier as the PRFs change from subswath to subswath. You may need to upsample the data to a fixed PRF so they can be combined later on for phase unwrapping.

**Reply by x3zou** [reply]
I was using Zeyu Jin's code for processing, as a 'black box'. I'd guess the part that performs resampling is:
 Could that be caused by the interpolation method parameter after the resamp command? I will try it out.
 <img width="721" alt="Screenshot 2023-03-07 at 11 11 14" src="https://user-images.githubusercontent.com/103137212/223526384-0bb1d4e7-203c-4cce-aff0-450600f2ad94.png">

update : It seems that changing the resamp interpolation doesn't really work. So I will try to use p2p_ALOS2_SCAN_Frame.csh

**Reply by Xiaohua-Eric-Xu** [reply]
Oops, that code does look familiar :). It looks fine to me and after resampling the PRMs should have the same num_rng_bins as in this line in resamp.c
```
162     ps.num_rng_bins = pm.num_rng_bins;
```
Is it OK for you to use p2p_ALOS2_SCAN_Frame.csh?  It'll be much easier for us to track done errors as we know all the controlling factors. The error may or maynot really come from this specific section.  If an advanced user (like Zeyu) take pieces of code and build a new processing chain, thing's usually gets really complicated.

**Reply by x3zou** [reply]
Yes, p2p_ALOS2_SCAN_Frame.csh works fine. No idea what happened in Zeyu's code. Anyway, thank you very much.

**Reply by Xiaohua-Eric-Xu** [reply]
It could be that we updated a bunch of modules while Zeyu's code got built with an earlier version of GMTSAR? This happens and is one of the reason sometimes folks do not want to update their code or system.
